src/VisualStudio/Core/Def/Implementation/Remote/JsonRpcConnection.cs

```diff
@@ -81,7 +81,7 @@ public override void Dispose()
             // Otherwise, dispose the underlying end-point and transition to "disposed" state.
 
             var poolReclamation = Interlocked.Exchange(ref _poolReclamation, null);
-            if (poolReclamation != null)
+            if (poolReclamation != null && !_serviceEndPoint.IsDisposed)
```

Comment:
i woudl be openly curious if we need this pooling at all.  @tmat do you know how necessary this is at all?

Reply:
I'm gonna delete it once we move to IServiceBroker and multiplexing streams.

Reply:
IDK if it adds much value currently. It was something that has been used since the beginning of OOP.